pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -267,10 +270,30 @@ func NewController(kubeClient kubelib.Client, options Options) *Controller {
 		})
 	})
 	registerHandlers(c.pods.informer, c.queue, "Pods", c.pods.onEvent, nil)
-
+	if err := updateSPIFFERegistry(kubeClient, c.clusterID); err != nil {
+		log.Errorf("Failed to update the trust domain for cluster %v: %v", c.clusterID, err)
+	}
 	return c
 }
 
+func updateSPIFFERegistry(kubeClient kubelib.Client, clusterID string) error {
+	cm, err := kubeClient.CoreV1().ConfigMaps(IstioNamespace).Get(context.TODO(), constants.IstioMeshConfigConfigMapName, metav1.GetOptions{})
```

Comment:
This seems incorrect on local cluster, what if the configmap and local file are different?

Reply:
obsolete.